app/src/main/java/com/bogdan/codeforceswatcher/features/main/BottomNavigationMenuFragment.kt

```diff
@@ -0,0 +1,117 @@
+package com.bogdan.codeforceswatcher.features.main
+
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.ViewGroup
+import androidx.compose.foundation.Image
+import androidx.compose.foundation.background
+import androidx.compose.foundation.clickable
+import androidx.compose.foundation.layout.*
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.material.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.MutableState
+import androidx.compose.runtime.State
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.graphics.ColorFilter
+import androidx.compose.ui.platform.ComposeView
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import androidx.fragment.app.Fragment
+import com.bogdan.codeforceswatcher.R
+import com.bogdan.codeforceswatcher.components.compose.theme.AlgoismeTheme
+
+class NavigationMenuFragment: Fragment() {
+
+    private val items = listOf(
+        MenuItem(MenuItem.Tab.CONTESTS, R.drawable.ic_contests, R.drawable.ic_contests_gradient),
+        MenuItem(MenuItem.Tab.USERS, R.drawable.ic_users, R.drawable.ic_users_gradient),
+        MenuItem(MenuItem.Tab.NEWS, R.drawable.ic_news, R.drawable.ic_news_gradient),
+        MenuItem(MenuItem.Tab.PROBLEMS, R.drawable.ic_problems, R.drawable.ic_problems_gradient),
+    )
+    private var selectedTab: MutableState<MenuItem.Tab> = mutableStateOf(MenuItem.Tab.CONTESTS)
+    lateinit var onMenuItemTap: (MenuItem.Tab) -> Unit
+
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ) = ComposeView(requireContext()).apply {
+        setContent {
+            AlgoismeTheme {
+                NavigationMenuView(items, selectedTab) {
+                    selectedTab.value = it
+                    onMenuItemTap(it)
+                }
+            }
+        }
+    }
+}
+
+@Composable
+private fun NavigationMenuView(
+    items: List<MenuItem>,
+    selectedTab: State<MenuItem.Tab>,
+    onMenuItemTap: (MenuItem.Tab) -> Unit
```

Comment:
Usually, for top-level composable we should pass a `modifier` (but not for other private non-top-level). It's mainly done for being able to configure different previews.